code/frameworks/angular/src/client/angular-beta/utils/PropertyExtractor.ts

```diff
@@ -0,0 +1,123 @@
+import { CommonModule } from '@angular/common';
+import { NgModuleMeta } from '@angular/compiler-cli/src/ngtsc/metadata';
+import { NgModule, Provider } from '@angular/core';
+import { NgModuleMetadata } from '../../types';
+import { isDeclarable, isStandaloneComponent } from './NgComponentAnalyzer';
+import { isComponentAlreadyDeclared } from './NgModulesAnalyzer';
+
+/**
+ * Flattens Array Deeply
+ *
+ * - Executes recursively
+ *
+ * @date `1/24/2023 - 12:58:03 PM`
+ * @author `sheriffMoose`
+ *
+ * @param {any[]} arr - Array to be flattened
+ * @returns {*} Array of imports
```

Comment:
Usually, we don't need this kind of JSDOC annotation; at least the type information is provided by Typescript. Again, let's only add descriptions for params and returns if it is crucial and not self-explaining.

Reply:
Sure